functorch/csrc/BatchRulesScatterOps.cpp

```diff
@@ -425,13 +425,41 @@ Tensor select_scatter_decomp(
   return at::scatter(self, dim, index_.expand_as(self), source.unsqueeze(dim).expand_as(self));
 }
 
+std::tuple<Tensor, optional<int64_t>> diagonal_scatter_batch_rule(
+    const Tensor &self, c10::optional<int64_t> self_bdim,
+    const Tensor &src, c10::optional<int64_t> src_bdim,
+    int64_t offset, int64_t dim1, int64_t dim2)
```

Comment:
is it possible to call diagonal_scatter in the batch rule?

Reply:
Ah that makes sense. Thanks!